app/src/main/java/org/mozilla/fenix/components/history/PagedHistoryProvider.kt

```diff
@@ -164,36 +162,13 @@ class DefaultPagedHistoryProvider(
      */
     suspend fun deleteMetadataSearchGroup(group: History.Group) {
         for (historyMetadata in group.items) {
-            getMatchingHistory(historyMetadata)?.let {
-                historyStorage.deleteVisit(
-                    url = it.url,
-                    timestamp = it.visitTime
-                )
-            }
+            historyStorage.deleteVisitsFor(historyMetadata.url)
```

Comment:
I think it's worth leaving a comment here on _why_ this actually works, and what our intention is here during a delete. Specifically, our intention now is that when user deletes anything using the History UI, we delete all occurrences of the deleted URLs from all of this browsing history.

May be worth leaving a technical note in the code, as well. Before we used to call an obviously-appropriate method named `deleteHistoryMetadata`, and now we do not. This relies on the ON DELETE CASCADE behaviour at the schema level within our metadata table in app services, something which is very much invisible at the application level.

Reply:
Well, you covered everything nicely, I have nothing left to add. Writing a tech note in code.